From 29fa062f67b017b82931776bc3f11c1e064f946e Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Mon, 11 Oct 2021 17:35:12 +0300
Subject: [PATCH 13/14] s32-gen1: Disable CONFIG_PRAM and remove unused code

Disabled the CONFIG_PRAM and removed the code that would
become unused.

Issue: ALB-7902
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/cpu.c |  4 ----
 arch/arm/cpu/armv8/s32/fdt.c | 26 --------------------------
 include/configs/s32-gen1.h   | 14 --------------
 3 files changed, 44 deletions(-)

diff --git a/arch/arm/cpu/armv8/s32/cpu.c b/arch/arm/cpu/armv8/s32/cpu.c
index 20f2ee10fa..a8127a3880 100644
--- a/arch/arm/cpu/armv8/s32/cpu.c
+++ b/arch/arm/cpu/armv8/s32/cpu.c
@@ -893,10 +893,6 @@ phys_size_t __weak get_effective_memsize(void)
 	s32_exclude_ecc_range(CONFIG_SYS_FSL_DRAM_BASE1, &size);
 #endif
 
-#ifdef CONFIG_PRAM
-	/* ATF space */
-	size -= CONFIG_PRAM * SZ_1K;
-#endif
 #endif
 	return size;
 }
diff --git a/arch/arm/cpu/armv8/s32/fdt.c b/arch/arm/cpu/armv8/s32/fdt.c
index b8e842cb45..c65acf13b6 100644
--- a/arch/arm/cpu/armv8/s32/fdt.c
+++ b/arch/arm/cpu/armv8/s32/fdt.c
@@ -243,28 +243,6 @@ static void hide_sram(bd_t *bd)
 	}
 }
 
-#if (defined(CONFIG_NXP_S32G2XX) || defined(CONFIG_NXP_S32G3XX)) && \
-	defined(CONFIG_PRAM)
-/* Fixup the DDR node in order to reserve "pram" amount of KB somewhere in the
- * available physical memory. This would typically be used by TF-A as a secure
- * memory, and enforced through XRDC. Making it "invisible" to Linux is only a
- * defensive means of keeping software out of trouble.
- * The point is, u-boot may not be able to probe the whole DRAM (and may not
- * care about all of it anyway), so using "mem=" bootargs would not be enough.
- */
-static void exclude_pram(bd_t *bd)
-{
-	int bank;
-
-	for (bank = 0; bank < CONFIG_NR_DRAM_BANKS; bank++) {
-		if (bd->bi_dram[bank].start == CONFIG_SYS_FSL_DRAM_BASE1) {
-			bd->bi_dram[bank].size -= CONFIG_PRAM * SZ_1K;
-			break;
-		}
-	}
-}
-#endif
-
 static void apply_memory_fixups(void *blob, bd_t *bd)
 {
 	u64 start[CONFIG_NR_DRAM_BANKS];
@@ -319,10 +297,6 @@ static void ft_fixup_memory(void *blob, bd_t *bd)
 {
 	hide_sram(bd);
 
-#if (defined(CONFIG_NXP_S32G2XX) || defined(CONFIG_NXP_S32G3XX)) && \
-	defined(CONFIG_PRAM)
-	exclude_pram(bd);
-#endif
 	apply_ddr_limits(bd);
 
 	apply_memory_fixups(blob, bd);
diff --git a/include/configs/s32-gen1.h b/include/configs/s32-gen1.h
index a760bd7cee..0a0d28e29d 100644
--- a/include/configs/s32-gen1.h
+++ b/include/configs/s32-gen1.h
@@ -130,18 +130,4 @@
 #define SDHC_REDUCED_MAP
 #endif
 
-#if !defined(CONFIG_PRAM) && defined(CONFIG_S32_ATF_BOOT_FLOW)
-
-/* 24 MB covering the following:
- *  - 22 MB for optee_os + shared memory between optee_os and linux kernel
- *  - 2 MB for the Secure Monitor
- */
-#define CONFIG_PRAM	24576	/* 24MB */
-
-#ifndef __ASSEMBLY__
-_Static_assert(CONFIG_PRAM * SZ_1K + INITRD_HIGH == INITRD_HIGH_DEFAULT,
-	       "Please adjust INITRD_HIGH when booting with TF-A");
-#endif
-
-#endif
 #endif
-- 
2.17.1

